### ISSUE TITLE
Attributes for spans?

### ISSUE THREAD

**Issue by rauschma** [arve0/markdown-it-attrs#38]
[Pandoc](http://pandoc.org/MANUAL.html#spans) supports the following syntax:

```
[This is *some text*]{.class key="val"}
```

It’d be lovely to have that as an option.

**Comment by arve0** [comment]
Working on this, but unsure how to do the token stream transform.

input
```
[span with *emph*]{.c}
```

token stream
```
[
  {
    "type": "paragraph_open",
    "tag": "p",
    "attrs": null,
    "map": [
      0,
      1
    ],
    "nesting": 1,
    "level": 0,
    "children": null,
    "content": "",
    "markup": "",
    "info": "",
    "meta": null,
    "block": true,
    "hidden": false
  },
  {
    "type": "inline",
    "tag": "",
    "attrs": null,
    "map": [
      0,
      1
    ],
    "nesting": 0,
    "level": 1,
    "children": [
      {
        "type": "text",
        "tag": "",
        "attrs": null,
        "map": null,
        "nesting": 0,
        "level": 0,
        "children": null,
        "content": "[span with ",
        "markup": "",
        "info": "",
        "meta": null,
        "block": false,
        "hidden": false
      },
      {
        "type": "em_open",
        "tag": "em",
        "attrs": null,
        "map": null,
        "nesting": 1,
        "level": 1,
        "children": null,
        "content": "",
        "markup": "*",
        "info": "",
        "meta": null,
        "block": false,
        "hidden": false
      },
      {
        "type": "text",
        "tag": "",
        "attrs": null,
        "map": null,
        "nesting": 0,
        "level": 1,
        "children": null,
        "content": "emph",
        "markup": "",
        "info": "",
        "meta": null,
        "block": false,
        "hidden": false
      },
      {
        "type": "em_close",
        "tag": "em",
        "attrs": null,
        "map": null,
        "nesting": -1,
        "level": 0,
        "children": null,
        "content": "",
        "markup": "*",
        "info": "",
        "meta": null,
        "block": false,
        "hidden": false
      },
      {
        "type": "text",
        "tag": "",
        "attrs": null,
        "map": null,
        "nesting": 0,
        "level": 0,
        "children": null,
        "content": "]{.c}",
        "markup": "",
        "info": "",
        "meta": null,
        "block": false,
        "hidden": false
      }
    ],
    "content": "[span with *emph*]{.c}",
    "markup": "",
    "info": "",
    "meta": null,
    "block": true,
    "hidden": false
  },
  {
    "type": "paragraph_close",
    "tag": "p",
    "attrs": null,
    "map": null,
    "nesting": -1,
    "level": 0,
    "children": null,
    "content": "",
    "markup": "",
    "info": "",
    "meta": null,
    "block": true,
    "hidden": false
  }
]
```

Might be possible to remove `[]{}` from `inline.content` and then reparse.

**Comment by Connum** [comment]
Any updates on this? I'm currently working on a project that needs to support spans with specific classes. I would love to come up with something myself to get this implemented asap, but I'm still in the process of trying to understand the internal mechanics of markdown-it and this plugin, so I don't think I'll be able to contribute much.

**Comment by arve0** [comment]
Hi @Connum! I've tried implementing, but it was not easily done. In gist, `[asdf]{.cls}` does not give a consistent pattern in token stream. The cleanest might therefor be to alter the parser of markdown-it.

**Comment by arve0** [comment]
Btw, @Connum, you could use `<span>` directly in your markdown, by turning on `html` in markdown-it:

```md
a paragraph with <span class="red">spanned *text*</span>
```

renders to

```html
<p>a paragraph with <span class="red">spanned <em>text</em></span></p>
```

**Comment by Connum** [comment]
Hi @arve0,

thanks for getting back to me so quickly and for creating the issue at markdown-it-span, a plugin I also looked at. I hope there'll be a solution soon. Regarding the HTML, the client of the project I'm working on specified that there should be no HTML allowed in the input. (It's kind of a CMS and they don't want end-users to mess anything up, so we need strict control about the allowed input and markup rules).

Kind regards
Constantin

**Comment by oshihirii** [comment]
For reference, I am using:
```
var md = require('markdown-it')({
html:  false,
xhtmlOut:  false, 
breaks:  false,
langPrefix:   'language-',  
linkify:  true,  
typographer:  false,
quotes: '“”‘’',
highlight: function (/*str, lang*/) { return ''; }
});

var markdownItAttrs = require('markdown-it-attrs');
var markdownItSpan = require('markdown-it-span');

md.use(markdownItAttrs).use(markdownItSpan);
```
Called with:
```
var new_value_rendered = md.renderInline(new_value);
```
Input:
```md
:the user menu:{.docs_heading}
```
Output:
```html
<span class="docs_heading">the user menu</span>
```

**Comment by mb21** [comment]
I've just created https://github.com/mb21/markdown-it-bracketed-spans

Seems to work quite well when used together with `markdown-it-attrs`. But currently, it create a span even on an invalid attribute, e.g. `[foo]{ bar` -> `<span>foo</span> bar`. Thus my question: does `markdown-it-attrs` expose a utility function that I can use? Similar to how  I used `state.md.helpers.parseLinkLabel`to parse the first part of the span...?
